controllers/ocimachine_controller.go

```diff
@@ -119,17 +116,29 @@ func (r *OCIMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
+	if len(ociCluster.Spec.Region) > 0 {
+		r.Region = ociCluster.Spec.Region
```

Comment:
this line is not correct right? why are we replacing the region for the reconciler? this wil create problems when the reconciler is reconciling clusters from 2 different regions right?